### DISCUSSION TITLE
@lingui/macro doesn't work with native ES modules

### DISCUSSION THREAD

**Discussion by fraction01** [lingui/js-lingui#1741] [Q&A] [answered]
Hi,

When I Try to use `@lingui/macro` with native ES modules I have this error:

![lingui macro esm issue](https://github.com/lingui/js-lingui/assets/14923003/2fd9406f-9062-442a-bd14-45342286f5f3)

```js
import express from 'express';
import { t } from "@lingui/macro";

const router = express.Router();

/* GET home page. */
router.get('/', function(req, res, next) {

  const hello = t`Hello World!`;
  res.render('index', { title: 'Express', hello });
});

export default router;
```

**To Reproduce**
- Clone [https://github.com/fraction01/lingui_macro_esm_issue](https://github.com/fraction01/lingui_macro_esm_issue) (originally generated with `npx express-generator` and converted to ESM)
- Run `npm start`

**Expected behavior**
Should work without problem.

**Additional context**

- node version ``16.19.1``
- jsLingui version ``4.3.0``
- Babel version ``@babel/core@7.22.9``
- Macro support:
 - [ ] I'm using SWC with `@lingui/swc-plugin`
 - [x] I'm using Babel with `babel-macro-plugin`
 - [ ] I'm not using macro
- Babel config (`.babelrc`):
```
{
    "plugins": ["macros"]
}
```

How can I get it work with native ES modules ?

Thank you

**Comment by timofei-iatsenko** [comment]
I didn't start your reproduction repo, but I bet the babel-macro-plugin just not configured correctly. Or babel not processing this exact file. So the problem not in the es modules but in your configuration.

**Reply by timofei-iatsenko** [reply, marked as answer]
I've just checked your example. You don't have babel preprocess step anywhere... You just run your js files directly without babel.

**Reply by fraction01** [reply]
Yes, you're right, I forgot to add `@babel/register` as in my app and I found the issue while reading the `@babel/register` [doc](https://babeljs.io/docs/babel-register#experimental-babel-8-implementation):
> **Note**: `@babel/register` does not support compiling native Node.js ES modules on the fly, since currently there is no stable API for intercepting ES modules loading.

Using `@babel/cli` works as expected, Thank you.
